gateway-ha/src/main/java/io/trino/gateway/ha/router/OAuth2GatewayCookieProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.AbstractModule;
+import com.google.inject.Inject;
+import io.airlift.units.Duration;
+import io.trino.gateway.ha.config.HaGatewayConfiguration;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class OAuth2GatewayCookieProvider
```

Comment:
I'm not sure why we need a new class as the logic is pretty simple. Why not putting `deletePaths` & `ttl` on fields in `ProxyRequestHandler`?

Reply:
We could do this. It seemed cleaner to keep the oauth2 cookie configuration out of `ProxyRequestHandler` since it does not use it directly. How do you typically draw the line between injecting vs plumbing configuration properties through by hand?